Gems/PhysX/Code/Tests/EditorColliderComponentTests.cpp

```diff
@@ -0,0 +1,138 @@
+
+namespace UnitTest
+{
+    class ColliderPickingFixture : public PhysXEditorColliderComponentManipulatorFixture
+    {
+    public:
+        void SetUpEditorFixtureImpl() override;
+        //! Clicks at a given position and returns the entities that are selected.
+        AzToolsFramework::EntityIdList ClickAndGetSelectedEntities(AzFramework::ScreenPoint screenPoint);
+
+        const float m_uniformScale = 1.0f;
+        const AZ::Quaternion m_shapeRotation = AZ::Quaternion(0.0f, 0, 0, 1);
+        const AZ::Quaternion m_entityRotation = AZ::Quaternion(0, 0, 0, 1);
+        const AZ::Vector3 m_shapeOffset = AZ::Vector3(0, 0, 0);
+        const AZ::Vector3 m_entityTranslation = AZ::Vector3(5.0f, 15.0f, 10.0f);
+    };
+
+    void ColliderPickingFixture::SetUpEditorFixtureImpl()
+    {
+        PhysXEditorColliderComponentManipulatorFixture::SetUpEditorFixtureImpl(); 
+
+        m_cameraState.m_viewportSize = AzFramework::ScreenSize(1920, 1080);
+        AzFramework::SetCameraTransform(
+            m_cameraState,
+            AZ::Transform::CreateFromQuaternionAndTranslation(
+                AZ::Quaternion::CreateFromEulerAnglesDegrees(AZ::Vector3(0.0f, 0.0f, 90.0f)), AZ::Vector3(20.0f, 15.0f, 10.0f)));
+
+         m_actionDispatcher->CameraState(m_cameraState);
+    }
+
+    AzToolsFramework::EntityIdList ColliderPickingFixture::ClickAndGetSelectedEntities(AzFramework::ScreenPoint screenPoint)
+    {
+        // click the entity in the viewport
+        m_actionDispatcher->SetStickySelect(true)
+            ->CameraState(m_cameraState)
+            ->MousePosition(screenPoint)
+            ->MouseLButtonDown()
+            ->MouseLButtonUp();
+
+        AzToolsFramework::EntityIdList selectedEntities;
+        AzToolsFramework::ToolsApplicationRequestBus::BroadcastResult(
+            selectedEntities, &AzToolsFramework::ToolsApplicationRequestBus::Events::GetSelectedEntities);
+        return selectedEntities;
+    }
+
+    TEST_F(ColliderPickingFixture, ColliderPickingWithBoxShape)
+    {
+         // Given the setup conditions
+         const AZ::Vector3 boxDimensions(5.0f, 5.0f, 5.0f);
+         SetupCollider(Physics::BoxShapeConfiguration(boxDimensions), m_shapeRotation, m_shapeOffset);
+         SetupTransform(m_entityRotation, m_entityTranslation, m_uniformScale);
+
+         // When a user clicks just outside the collider it should not be selected
+         auto clickPos1 = AzFramework::WorldToScreen(AZ::Vector3(7.5f, 12.4f, 10.0f), m_cameraState);
+         auto selectedEntities = ClickAndGetSelectedEntities(clickPos1);
+
+         EXPECT_THAT(selectedEntities.size(), testing::Eq(0));
+
+         // Then when a user clicks inside the collider it should be selected
+         auto clickPos2 = AzFramework::WorldToScreen(AZ::Vector3(7.5f, 12.6f, 10.0f), m_cameraState);
+         selectedEntities = ClickAndGetSelectedEntities(clickPos2);
+
+         EXPECT_THAT(selectedEntities.size(), testing::Eq(1));
```

Comment:
suggestion: Might be worth checking the entity id itself definitely matches too 🙂 (repeated)

Reply:
Thanks!